pytorch_lightning/trainer/trainer.py

```diff
@@ -1249,3 +1251,11 @@ def _log_device_info(self) -> None:
                 "IPU available but not used. Set the `ipus` flag in your trainer"
                 " `Trainer(ipus=8)` or script `--ipus=8`."
             )
+
+    def _on_expection(self):
+        if not self.is_global_zero or not _fault_tolerant_enabled():
+            return
+
+        # save a checkpoint for fault tolerant training
+        file_path = os.path.join(self.default_root_dir, ".pl_auto_save.ckpt")
```

Comment:
Why the period before the name? to hide it from the users?

Reply:
open to suggestion for naming. 
it depends how Lightning will use this checkpoint. Will it be fully automated resuming? Or does the user interact with this file to manually restart?

we may also need to provide a rotating filename